examples/e2e-test-dataflow/src/test/java/e2etest/E2ETest.java

```diff
@@ -0,0 +1,212 @@
+        String gcsbucket= myDataTable.get("inputFilePattern");
+        String bucket = "gs://" + gcsbucket.split("/")[2];
+        String command= String.format(string, bucket);
+        BashOutput output = MyBashExecutor.executeCommand(command);
+        assert (output.getStatus() == 0);
+
+    }
+
+    @Then("Copy {string} to inputFilePattern")
+    public void copy_to_inputFilePattern(String files)  throws Exception{
+        // Write code here that turns the phrase above into concrete actions
+        String inputPattern=myDataTable.get("inputFilePattern");
+        String bucket=inputPattern.substring(0,inputPattern.lastIndexOf("/"));
+        BashOutput output = Utils.copyFileFromLocalToGCS(bucket,files);
+        assert (output.getStatus() == 0);
+
+    }
+
+    @Then("Create tempLocation if not present")
+    public void create_tempLocation_if_not_present()  throws Exception{
+        String gcsbucket= myDataTable.get("tempLocation");
+        String bucket = "gs://" +  gcsbucket.split("/")[2];
+        BashOutput output = Utils.createGCSBucket(bucket);
+        assert (output.getStatus() == 0);
+    }
+
+    @When("Run dataflow application main class being {string} with proper parameters")
+    public void run_dataflow_application_main_class_being_with_proper_parameters(String mainClass)  throws Exception{
+
+        StringBuffer sb = new StringBuffer();
+        String[] feildForDataflowParams = {"tempLocation","project","serviceAccount","runner","dataset","tableName","inputFilePattern"};
+        for (String key : feildForDataflowParams) {
+            sb.append("--").append(key).append("=").append(myDataTable.get(key)).append(" ");
+        }
+        String command = "java -cp data/code/" + jarName + " " + mainClass + " " + sb.toString();
+        BashOutput output = MyBashExecutor.executeCommand(command);
+
+        for (String s : output.getOutput()){
+            if(s.contains("Submitted job:")){
+                jobId = s.replace("Submitted job:", "").trim();
+            }
+        }
+        assert (output.getStatus() == 0);
+    }
+
+    @Then("Wait for the Dataflow application to complete, maximum wait time {int} minutes")
+    public void wait_for_the_dataflow_application_to_complete_maximum_wait_time_minutes(Integer time) throws Exception{
+        String doneStateRunning="currentState: JOB_STATE_RUNNING";
+        String doneStatePending="currentState: JOB_STATE_PENDING";
+        String doneStateDone="currentState: JOB_STATE_DONE";
+        long startTime= System.currentTimeMillis();
+        String command = String.format("gcloud dataflow jobs describe %s", jobId);
+        BashOutput result;
+        do {
+            result = MyBashExecutor.executeCommand(command);
+            System.out.println(result.getOutput().get(2));
+            Thread.sleep(10000L);
+        } while ((Utils.checkIfStringIsInList(result.getOutput(), doneStateRunning) || Utils.checkIfStringIsInList(result.getOutput(), doneStatePending)) &&
```

Comment:
result.getOutput() returns a type of List<String> which has method contains to check if String exists or not. checkIfStringIsInList method can be avoided using library method